From ff25d1896d3befe184aa1adebf5d5e961f5d4b15 Mon Sep 17 00:00:00 2001
From: antirez <antirez@gmail.com>
Date: Fri, 5 Dec 2014 10:51:09 +0100
Subject: [PATCH] Simplify lua_cmsgpack macro and fix build on old Linux
 distros.

Thanks to @badboy for the help in checking the build after the fix.
---
 deps/lua/src/lua_cmsgpack.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/deps/lua/src/lua_cmsgpack.c b/deps/lua/src/lua_cmsgpack.c
index 6aa04e2..4ccf07f 100644
--- a/deps/lua/src/lua_cmsgpack.c
+++ b/deps/lua/src/lua_cmsgpack.c
@@ -18,14 +18,8 @@
     #define LUACMSGPACK_MAX_NESTING  16 /* Max tables nesting. */
 #endif
 
-#if (_XOPEN_SOURCE >= 600 || _ISOC99_SOURCE || _POSIX_C_SOURCE >= 200112L)
-    #define IS_FINITE(x) isfinite(x)
-#else
-    #define IS_FINITE(x) ((x) == (x) && (x) + 1 > (x))
-#endif
-
 /* Check if float or double can be an integer without loss of precision */
-#define IS_INT_TYPE_EQUIVALENT(x, T) (IS_FINITE(x) && (T)(x) == (x))
+#define IS_INT_TYPE_EQUIVALENT(x, T) (!isinf(x) && (T)(x) == (x))
 
 #define IS_INT64_EQUIVALENT(x) IS_INT_TYPE_EQUIVALENT(x, int64_t)
 #define IS_INT_EQUIVALENT(x) IS_INT_TYPE_EQUIVALENT(x, int)
